apps/meteor/tests/e2e/omnichannel/omnichannel-agents.spec.ts

```diff
@@ -1,44 +1,104 @@
+
+import { ILivechatDepartment } from '@rocket.chat/core-typings';
+
+import { IS_EE } from '../config/constants';
 import { Users } from '../fixtures/userStates';
 import { OmnichannelAgents } from '../page-objects';
+import { createDepartment, deleteDepartment } from '../utils/omnichannel/departments';
 import { test, expect } from '../utils/test';
 
 test.use({ storageState: Users.admin.state });
 
 test.describe.serial('omnichannel-agents', () => {
 	let poOmnichannelAgents: OmnichannelAgents;
 
-	test.beforeEach(async ({ page }) => {
-		poOmnichannelAgents = new OmnichannelAgents(page);
+	let department: ILivechatDepartment;
 
+	
+	test.beforeEach(async ({ api, page }) => {
+		poOmnichannelAgents = new OmnichannelAgents(page);
 		await page.goto('/omnichannel');
 		await poOmnichannelAgents.sidenav.linkAgents.click();
+		const res = await createDepartment(api);
+
+		await expect(res.status()).toBe(200);
+
+		({ department } = await res.json())
 	});
 
-	test('expect add "user1" as agent', async ({ page }) => {
+	test.afterEach(async ({ api }) => {
+		// Ensure that there is no leftover data even if test fails
+		await api.delete('/livechat/users/agent/user1');
+		await api.post('/settings/Omnichannel_enable_department_removal', { value: true }).then((res) => expect(res.status()).toBe(200));
+		await deleteDepartment(api, { id: department._id });
+		await api.post('/settings/Omnichannel_enable_department_removal', { value: false }).then((res) => expect(res.status()).toBe(200));
+	});
+
+	test('OC - Manage Agents - Add, search and remove using table', async ({ page }) => {
+		await test.step('expect "user1" be first ', async () => {
+			await poOmnichannelAgents.inputUsername.type('user');
+			await expect(page.locator('role=option[name="user1"]')).toContainText('user1');
+
+			await poOmnichannelAgents.inputUsername.type('');
+		});
```

Comment:
This step seems redundant considering the next step does almost exactly the same actions.